src/server/saved_objects/service/lib/repository.js

```diff
@@ -353,9 +361,6 @@ export class SavedObjectsRepository {
           type,
           ...time && { updated_at: time },
           version: doc._version,
-          ...extraDocumentProperties
```

Comment:
Previously, `extraDocumentProperties` was used to retrieve the `spaceId` (or now, `namespace`) property in order to verify that the object belongs to the current space.

Now, we are still restricting to the current space, but we aren't using this property to do so. We are using the composite document id to perform this check instead. This is more efficient than what I had, but it deviates from the property being the authoritative source of this information. In other words, we aren't using the `namespace` property for `get` or `bulk_get` operations anymore. By extension, this property isn't being used to verify `update` or `delete` operations, either.

I'm not necessarily opposed to changing our direction, but if I understand this right, the `namespace` property is only used for `find` at this point, which feels inconsistent. What are your thoughts?

Reply:
I shared your hesitation when initially making these changes, and it does remove the `namespace` property itself being the authoritative source of this information. However, it's equivalent to how we're using the `type` within the Repository itself. When in the context of `delete`\`get`\`update` we use the type and the namespace prepended to the IDs as the source of this information; however, the `find` operation relies upon the `type` and `namespace` attributes.

Reply:
That makes sense -- thanks for drawing the parallel to how we use the `type` property. That convinced me!